### PR TITLE
feat: integrate mach exchange

### PR DESCRIPTION
**Pull request by 4812571**

### CODE REVIEW

**Comment by llamatester** [comment]
The mach adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mach
    
    🦙 Running MACH adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 20 Nov 2024 06:03:40 GMT
    End Date:	Thu, 21 Nov 2024 06:03:40 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 23/8/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    OPTIMISM 👇
    Backfill start time: 14/11/1973
    NO METHODOLOGY SPECIFIED
    Daily volume: 204.06 k
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    ARBITRUM 👇
    Backfill start time: 22/3/1977
    NO METHODOLOGY SPECIFIED
    Daily volume: 218.89 k
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    AVAX 👇
    Backfill start time: 4/7/1971
    NO METHODOLOGY SPECIFIED
    Daily volume: 1
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    POLYGON 👇
    Backfill start time: 13/11/1971
    NO METHODOLOGY SPECIFIED
    Daily volume: 1
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    BASE 👇
    Backfill start time: 11/7/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 92.22 k
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    CELO 👇
    Backfill start time: 3/11/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    BSC 👇
    Backfill start time: 10/4/1971
    NO METHODOLOGY SPECIFIED
    Daily volume: 10.44 k
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    OP_BNB 👇
    Backfill start time: 29/11/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    MANTLE 👇
    Backfill start time: 6/2/1972
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    SCROLL 👇
    Backfill start time: 26/3/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    MODE 👇
    Backfill start time: 28/4/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)
    
    
    BLAST 👇
    Backfill start time: 6/3/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732169019 (2024-11-21T06:03:39.000Z)

File: dexs/mach/deployments.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+type Deployment = {
+    address: string,
+    start: number,
+}
+
+const DEPLOYMENTS: { [chain: string]: Deployment } = {
+    [CHAIN.ETHEREUM]: {
+        address: "0x5d8bca5F0b3D9c9513a75D0206dAF0b4FF8bda95",
+        start: 20299258,
```
Comment (g1nt0ki) [inline dexs/mach/deployments.ts]: thanks for the PR, the code is mostly good except for this part, this needs to unix timestamp or string like `'2023-02-03'`

Comment (g1nt0ki) [reply dexs/mach/deployments.ts]: also, removed the env file, think you were leaking your api key

**Comment by llamatester** [comment]
The mach adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mach
    
    🦙 Running MACH adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 20 Nov 2024 14:06:25 GMT
    End Date:	Thu, 21 Nov 2024 14:06:25 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 23/8/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    OPTIMISM 👇
    Backfill start time: 14/11/1973
    NO METHODOLOGY SPECIFIED
    Daily volume: 6.62 k
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    ARBITRUM 👇
    Backfill start time: 22/3/1977
    NO METHODOLOGY SPECIFIED
    Daily volume: 36.00 k
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    AVAX 👇
    Backfill start time: 4/7/1971
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.15 k
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    POLYGON 👇
    Backfill start time: 13/11/1971
    NO METHODOLOGY SPECIFIED
    Daily volume: 28
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    BASE 👇
    Backfill start time: 11/7/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 26.75 k
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    CELO 👇
    Backfill start time: 3/11/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    BSC 👇
    Backfill start time: 10/4/1971
    NO METHODOLOGY SPECIFIED
    Daily volume: 10.33 k
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    OP_BNB 👇
    Backfill start time: 29/11/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    MANTLE 👇
    Backfill start time: 6/2/1972
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    SCROLL 👇
    Backfill start time: 26/3/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    MODE 👇
    Backfill start time: 28/4/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)
    
    
    BLAST 👇
    Backfill start time: 6/3/1970
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732197984 (2024-11-21T14:06:24.000Z)

**Comment by 4812571** [comment]
@g1nt0ki This should fix all of your concerns.

**Comment by llamatester** [comment]
The mach adapter exports:
        
 
 
    > adapters@1.0.0 test
    > ts-node --transpile-only cli/testAdapter.ts dexs mach
    
    🦙 Running MACH adapter 🦙
    ---------------------------------------------------
    Start Date:	Wed, 20 Nov 2024 17:17:08 GMT
    End Date:	Thu, 21 Nov 2024 17:17:08 GMT
    ---------------------------------------------------
    
    ETHEREUM 👇
    Backfill start time: 13/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    OPTIMISM 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 444
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    ARBITRUM 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 37.84 k
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    AVAX 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 1.15 k
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    POLYGON 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 28
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    BASE 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 13.35 k
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    CELO 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    BSC 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 10.32 k
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    OP_BNB 👇
    Backfill start time: 9/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    MANTLE 👇
    Backfill start time: 9/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    SCROLL 👇
    Backfill start time: 9/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    MODE 👇
    Backfill start time: 9/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)
    
    
    BLAST 👇
    Backfill start time: 2/7/2024
    NO METHODOLOGY SPECIFIED
    Daily volume: 0
    End timestamp: 1732209427 (2024-11-21T17:17:07.000Z)

**Pull request merged**